shared-libs/message-utils/src/index.js

```diff
@@ -96,21 +96,52 @@ const applyPhoneFilters = function(config, phone) {
   return phone;
 };
 
+const normalizeRecipient= function(recipient) {
+  if (Array.isArray(recipient)) {
+    return recipient.map(r => typeof r === 'string' ? r.trim() : r);
+  }
+
+  return typeof recipient === 'string'
+    ? recipient.trim()
+    : recipient;
+};
+
 const getRecipient = function(context, recipient, defaultToSender = true) {
   if (!context) {
     return;
   }
 
   const from = context.from || context.contact?.phone;
-  recipient = recipient?.trim();
+  recipient = normalizeRecipient(recipient);
 
   if (!recipient) {
     return from;
   }
 
-  const phone = resolveRecipient(context, recipient);
+  const phone = resolveMany(context, recipient);
+  const fallBackRecipient = Array.isArray(recipient) ? recipient[0]: recipient;
+  return phone || (defaultToSender && from) || fallBackRecipient;
```

Comment:
`recipient` would always be an array: 
```suggestion
  return phone || (defaultToSender && from) || recipient[0];
```